server/app/services/applications/ProgramAdminApplicationService.java

```diff
@@ -53,16 +84,90 @@ public Optional<Application> getApplication(long applicationId, ProgramDefinitio
+    Applicant applicant = application.getApplicant();
+    String newStatusText = newStatusEvent.statusText();
+    // The send/sent phrasing is a little weird as the service layer is converting between intent
+    // and reality.
+    boolean sendEmail = newStatusEvent.emailSent();
+
+    Optional<Status> statusDefMaybe =
+        program.getStatusDefinitions().getStatuses().stream()
+            .filter(s -> s.statusText().equals(newStatusText))
+            .findFirst();
+    if (statusDefMaybe.isEmpty()) {
+      throw new StatusNotFoundException(newStatusText, program.id);
+    }
+    Status statusDef = statusDefMaybe.get();
+
     ApplicationEventDetails details =
         ApplicationEventDetails.builder()
             .setEventType(ApplicationEventDetails.Type.STATUS_CHANGE)
-            .setStatusEvent(newStatus)
+            .setStatusEvent(newStatusEvent)
             .build();
     ApplicationEvent event = new ApplicationEvent(application, admin, details);
+
+    // Send email if requested and present.
+    if (sendEmail && statusDef.localizedEmailBodyText().isPresent()) {
+      // Notify an Admin/TI if they applied.
+      Optional<String> adminSubmitterEmail = application.getSubmitterEmail();
+      if (adminSubmitterEmail.isPresent()) {
+        sendAdminSubmitterEmail(programName, applicant, newStatusText, adminSubmitterEmail);
+      }
+      // Notify the applicant.
+      Optional<String> applicantEmail =
+          applicantService.getEmail(application.getApplicant().id).toCompletableFuture().join();
+      if (applicantEmail.isPresent()) {
+        sendApplicantEmail(programName, applicant, statusDef, applicantEmail);
+      }
+    }
+
     eventRepository.insertSync(event);
   }
 
+  private void sendApplicantEmail(
+      String programName, Applicant applicant, Status statusDef, Optional<String> applicantEmail) {
+    String civiformLink = baseUrl;
+    Locale locale = applicant.getApplicantData().preferredLocale();
+    String emailBody =
+        String.format(
+            "%s\n\nLog in to CiviForm at %s.",
+            statusDef.localizedEmailBodyText().get().getOrDefault(locale), civiformLink);
+    emailClient.send(
+        isStaging ? stagingApplicantNotificationMailingList : applicantEmail.get(),
+        String.format(STATUS_UPDATE_EMAIL_SUBJECT_FORMAT, programName),
+        emailBody);
+  }
+
+  private void sendAdminSubmitterEmail(
+      String programName,
+      Applicant applicant,
+      String newStatusText,
+      Optional<String> adminSubmitterEmail) {
+    String tiDashLink =
+        baseUrl
+            + controllers.ti.routes.TrustedIntermediaryController.dashboard(
+                    /* nameQuery= */ Optional.empty(),
+                    /* dateQuery= */ Optional.empty(),
+                    /* page= */ Optional.of(1))
+                .url();
+    String subject =
+        String.format(
+            "An update on the application for program %s on behalf of applicant %d",
```

Comment:
I'd either put the applicant name as the last argument or change the text a smidge since "on behalf of applicant 2" doesn't read very clearly to me.

Reply:
OK.